src/components/PopupWithForm.js

```diff
@@ -35,14 +39,29 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  _renderLoading(isLoading) {
+    this._submitButton.textContent = isLoading
+      ? "Saving..."
+      : this._submitButtonText;
+  }
```

Comment:
**COULD BE IMPROVED**


`"Saving..."` is hardcoded here

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```